src/healthcheck.ts

```diff
@@ -4,17 +4,24 @@ import http from 'http';
 const app = express();
 const router = express.Router();
 
-router.use((req, res, next) => {
+router.use((req: express.Request, res: express.Response, next: express.NextFunction) => {
 	res.header('Access-Control-Allow-Methods', 'GET');
-	next();
+	const status = app.locals.getStatus()
+	if (status === 200) {
+		next();
+	} else {
+		res.status(status).send({ error: 'Something failed!' })
+	}
 });
 
-router.get('/health', (req, res) => {
-	res.status(200).send('Ok');
+router.get('/health', (req: express.Request, res: express.Response) => {	
+		res.status(200).send('Ok');
 });
 
 app.use('/api/v1', router);
 
-const server = http.createServer(app);
-
-export default server.listen(3000);
+export default function (callback: any) {
```

Comment:
The callback method has no signature. And it's implementation returns either 200 or 502. Which we're then putting back into an if statement. Would it make sense to instead have a `() => bool` that just wraps the ready call?

Reply:
Slight nitpick: generally I think named arguments are helpful so on the topic of argument types an object with the callback may be nice :]